### ISSUE TITLE
Cannot find module 'pouchdb/extras/promise'

### ISSUE THREAD

**Issue by denis-sokolov** [nolanlawson/pouchdb-authentication#25]
f76b4495 breaks the library. [Offending code](https://github.com/nolanlawson/pouchdb-authentication/blob/f76b4495fcb83b4943927a5175d2aace1d24ff86/lib/utils.js#L3):

```javascript
var Promise = require('pouchdb/extras/promise');
```
```
module.js:318
    throw err;
    ^
Error: Cannot find module 'pouchdb/extras/promise'
    at Function.Module._resolveFilename (module.js:316:15)
    at Function.Module._load (module.js:258:25)
    at Module.require (module.js:345:17)
    at require (module.js:364:17)
    at Object.<anonymous> (.../node_modules/pouchdb-authentication/lib/utils.js:3:15)
```

This assumes that `pouchdb` installed is the latest version. But this is neither required by [the `package.json`](https://github.com/nolanlawson/pouchdb-authentication/blob/f76b4495fcb83b4943927a5175d2aace1d24ff86/package.json#L34) (use a [commit-ish suffix](https://docs.npmjs.com/files/package.json#git-urls-as-dependencies)), neither makes direct sense as the `pouchdb-authentication` depends on its own `pouchdb`, which may be different from the `pouchdb` my application depends on.
The dependency between `pouchdb-authentication` and `pouchdb` is more of a [peer dependency](https://docs.npmjs.com/files/package.json#peerdependencies), not a direct dependency.

My suggested architecture for plugins is to take the host object as a parameter in the initialization of the plugin. Which versions of the host object are accepted is then part of a public API of the plugin.
Then, if the plugin one day requires a new feature from a new version of the host object, it's a breaking change in the plugin and requires a major version bump.

**Comment by nolanlawson** [comment]
(Related to https://github.com/pouchdb/pouchdb/pull/3730.)

My original goal was for pouchdb-authentication to depend on latest pouchdb, regardless of the version of pouchdb the user of pouchdb-authentication might have. Since npm allows dependencies inside of dependencies, I figured this would be fine. At the end of the day, either you'll have the same version of PouchDB, in which case browserify will dedup, or you'll have a slightly different version, in which case the code will probably be the same anyway, so gzip will dedup.

My understanding was that [this line](https://github.com/nolanlawson/pouchdb-authentication/blob/f76b4495fcb83b4943927a5175d2aace1d24ff86/package.json#L34) would fetch the latest pouchdb from Github. Is there a better way?

**Comment by denis-sokolov** [comment]
I am not sure. From what I initially observed it seems that `npm` has deduped and decided that my `3.3.1` will be a good match for your `pouchdb/pouchdb`. I will experiment and confirm whether that's right.

Is there a reason you have not declared your dependency as `^3.4.0`?

**Comment by nolanlawson** [comment]
yeah, the `extras` API is not released yet. It will be released in 3.5.0.

Maybe if we change it to a literal `https://github.com/pouchdb/pouchdb.git#master` or something? could you try that?

**Comment by denis-sokolov** [comment]
Hm. I am unable to reproduce this again.
Even a manual `npm dedupe` results in the following:
```
npm WARN unavoidable conflict pouchdb { '.../node_modules/pouchdb': '3.4.0',
npm WARN unavoidable conflict   '.../node_modules/pouchdb-authentication/node_modules/pouchdb': '3.4.1-prerelease' }
npm WARN unavoidable conflict Not de-duplicating
```
Perhaps something has changed. Oh well.
